base/io.jl

```diff
@@ -564,7 +564,23 @@ function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
     end
 end
 
+const _os = ccall(:jl_get_UNAME, Any, ())
+
 function write(io::IO, c::Char)
+@static if _os === :Windows || _os === :NT
```

Comment:
I think we don't need to seperate Windows and others. Using `Ref` worked well on the others, so I think using `fill` will also be okay.

Reply:
How about replacing `while` with `unsafe_write` for other platforms by removing `if _os === :Windows || _os === :NT` statement. (Of course, if `unsafe_write` is faster.) Eventually this implementation may go into `string(::Char)` and we use `write(io::IO, c::Char) = write(io, string(c))`.